Puzzling/Puzzling/Presentation/CreateProject/ViewController/CreateProjectViewController.swift

```diff
@@ -92,4 +111,137 @@ extension CreateProjectViewController {
+    }
+    
+    private func setDelegate() {
+        createProjectView.startDayView.projectStartDelegate = self
+    }
+    
+    private func setNotificationCenter() {
+        NotificationCenter.default.addObserver(self, selector: #selector(getTextFieldInfo(_:)), name: Notification.Name("textFieldNotification"), object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(getStartDateInfo(_:)), name: Notification.Name("startDateNotification"), object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(getCycleInfo(_:)), name: Notification.Name("cycleNotification"), object: nil)
+    }
+    
+    func presentToHalfModalViewController() {
+        let projectStartTimeVC = ProjectStartTimeViewController()
+        projectStartTimeVC.modalPresentationStyle = .pageSheet
+        let screenHeight = UIScreen.main.bounds.height
+        let customDetentIdentifier = UISheetPresentationController.Detent.Identifier("customDetent")
+        let customDetent = UISheetPresentationController.Detent.custom(identifier: customDetentIdentifier) { (_) in
+            if screenHeight > 670 {
+                return 311
+            } else {
+                return 330
+            }
+        }
+        
+        if let sheet = projectStartTimeVC.sheetPresentationController {
+            sheet.detents = [customDetent]
+            sheet.preferredCornerRadius = 16
+            sheet.delegate = self
+            sheet.prefersGrabberVisible = true
+        }
+        present(projectStartTimeVC, animated: true, completion: nil)
+    }
+    
+    private func projectRegister() {
+        print("registerProjectButtonTap!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+        print(projectName)
+        print(projectDescription)
+        print(projectStartDate)
+        print(projectRole)
+        print(projectNickname)
+        print(projectCycle)
+    }
+    
+    private func getSelectedProjectCycle(list: [Int]) -> [String] {
+        var selectedList = list
+        selectedList.sort()
+        let dayOfWeekList = selectedList.map { index in
+            switch index {
+            case 0:
+                return "월"
+            case 1:
+                return "화"
+            case 2:
+                return "수"
+            case 3:
+                return "목"
+            case 4:
+                return "금"
+            case 5:
+                return "토"
+            case 6:
+                return "일"
+            default:
+                return ""
+            }
+        }
+        return dayOfWeekList
+    }
+    
+    private func buttonStateSetting() {
+        if (!projectName.isEmpty && !projectDescription.isEmpty && !projectStartDate.isEmpty &&
+            !projectRole.isEmpty && !projectNickname.isEmpty && !projectCycle.isEmpty) {
+            registerProjectButton.setState(.allow)
+        } else {
+            registerProjectButton.setState(.notAllow)
+        }
```

Comment:
```suggestion
    let projectInfoFields = [projectName, projectDescription, projectStartDate, projectRole, projectNickname, projectCycle]
    let isValidInput = projectInfoFields.contains { $0.isEmpty }

    let buttonState: ButtonState = isValidInput ? .notAllow : .allow
    registerProjectButton.setState(buttonState)
```

정도로 조금 간결하게 표현 해볼 수는 있을 것 같습니다!

Reply:
와 감사합니다 !!!!!